File: Editorial/3001-3100/3062_Winner_of_the_Linked_List_Game/Java/2_Point_Difference.java
class Solution {
    public String gameResult(ListNode head) {
        ListNode currentNode = head;
        int pointDifference = 0;

        while (currentNode != null) {
            // Update the point difference based on the comparison of current and next nodes
            if (currentNode.next != null) {
                pointDifference += (currentNode.val > currentNode.next.val) ? 1 : -1;
            }
            
            // Move two steps ahead in the linked list to the next even node
            currentNode = currentNode.next.next;
        }

        // Determine the winner based on the final score difference
        if (pointDifference < 0)
            return "Odd";
        else if (pointDifference > 0)
            return "Even";
        else
            return "Tie";
    }
}

File: Editorial/3001-3100/3062_Winner_of_the_Linked_List_Game/Python3/2_Point_Difference.py
class Solution:
    def gameResult(self, head: Optional[ListNode]) -> str:
        current_node, point_difference = head, 0

        while current_node:
            # Update the point difference based on the comparison of current and next nodes
            point_difference += (current_node.val > current_node.next.val) - (current_node.val < current_node.next.val)
            
            # Move two steps ahead in the linked list to the next even node
            current_node = current_node.next.next

        # Determine the winner based on the final score difference
        if point_difference < 0:
            return "Odd"
        elif point_difference > 0:
            return "Even"
        else:
            return "Tie"
